core/src/main/java/org/jclouds/rest/internal/RestAnnotationProcessor.java

```diff
@@ -573,7 +573,11 @@ static URI addHostIfMissing(URI original, URI withHost) {
          return null;
       if (original.getHost() != null)
          return original;
-      return withHost.resolve(original);
+      URI baseURI  = withHost;
```

Comment:
Better use an inline if to avoid creating a variable just to reassign it immediately after.